src/libutil/serialise.hh

```diff
@@ -501,4 +501,14 @@ struct StackAllocator {
     static StackAllocator *defaultAllocator;
 };
 
+/* Disabling GC when entering a coroutine (on macos).
+   ::create is to avoid boehm gc dependency in libutil.
+ */
+class DisableGC {
```

Comment:
Could we maybe make this a slightly more generic `CoroutineContext` or something like that? Just so that
- this layer doesn't have to bother about the GC
- Rather than having all these `#if __APPLE__` in `serialise.cc` we can just unconditionally call `CoroutineContext.create()` and put this `#if` inside the constructor/destructor of the `BoehmDisableGC` class

Reply:
Good idea, I do want to keep it obvious what it's doing and why, though. I've added the disablegc call inside the coroutine definition now.
I considered having a generic hook system (with std::list and everything) so anywhere in the code can add arbitrary coroutine context hooks, but this seems overkill, slower and less clear.